src/main/java/bio/terra/service/filedata/flight/ingest/IngestCopyLoadHistoryStep.java

```diff
@@ -50,48 +54,72 @@ public StepResult doStep(FlightContext context) {
     UUID datasetId = UUID.fromString(datasetIdString);
     Dataset dataset = datasetService.retrieve(datasetId);
 
-    int chunkNum = 0;
-    List<BulkLoadHistoryModel> loadHistoryArray = null;
     String flightId = context.getFlightId();
     String tableName_FlightId = flightId.replaceAll("[^a-zA-Z0-9]", "_");
     try {
       Instant loadTime = context.getStairway().getFlightState(flightId).getSubmitted();
-      bigQueryPdao.createStagingLoadHistoryTable(dataset, tableName_FlightId);
-      TimeUnit.SECONDS.sleep(waitSeconds);
+      LoadHistoryIterator historyIterator = loadService.loadHistoryIterator(loadId, fileChunkSize);
 
-      while (loadHistoryArray == null || loadHistoryArray.size() == fileChunkSize) {
-        loadHistoryArray =
-            loadService.makeLoadHistoryArray(loadId, fileChunkSize, (chunkNum * fileChunkSize));
-        chunkNum++;
-        // send list plus load_tag, load_time to BQ to be put in a staging table
-        if (!loadHistoryArray.isEmpty()) {
-          bigQueryPdao.loadHistoryToStagingTable(
-              dataset, tableName_FlightId, loadTag, loadTime, loadHistoryArray);
-          // Sleep to avoid BQ rate limit error
-          // From quick survey of logs, longest time to complete load query: 3 seconds
-          TimeUnit.SECONDS.sleep(waitSeconds);
-        }
+      if (platform.isGcp()) {
+        doGcpStep(dataset, tableName_FlightId, loadTime, historyIterator);
+      } else {
```

Comment:
I'm thinking that it's safer to do:
```suggestion
      } else if (platform.isAzure()) {
```